include/dr/mhp/containers/distribution.hpp

```diff
@@ -22,6 +22,11 @@ struct distribution {
     return *this;
   }
 
+  distribution &halo(std::size_t prev, std::size_t next) {
```

Comment:
same signature as previous function, it can not compile (see CI)

also you overwrite here periodic parameter by assigning it false

please remove this function